pmd-cpp/src/main/java/net/sourceforge/pmd/cpd/CPPTokenizer.java

```diff
@@ -92,4 +69,14 @@ private String maybeSkipBlocks(String test) throws IOException {
         }
         return filtered.toString();
     }
+
+    @Override
+    protected TokenManager getLexerForSource(SourceCode sourceCode) {
+        try {
+            StringBuilder buffer = sourceCode.getCodeBuffer();
+            return new CppTokenManager(IOUtil.skipBOM(new StringReader(maybeSkipBlocks(buffer.toString()))));
```

Comment:
we should add a TODO to move `maybeSkipBlocks` logic into a tokenfilter extending `JavaCCTokenFilter`